packages/near-lake-primitives/src/types/block.ts

```diff
@@ -0,0 +1,159 @@
+import { Action, isActionReceipt, outcomeWithReceiptToReceipt, receiptViewToAction } from './receipts';
+import { StreamerMessage, ValidatorStakeView } from './core/types';
+import { Receipt } from './receipts';
+import { Transaction } from './transactions';
+import { Event, logToRawEvent } from './events';
+import { fromStateChangeViewToStateChange, StateChange } from './state_changes';
+
+export class Block {
+    readonly streamerMessage: StreamerMessage;
+    private executedReceipts: Array<Receipt>;
+    readonly postponedReceipts: Array<Receipt>;
+    readonly transactions: Array<Transaction>;
+    private _actions: Map<string, Action>;
+    private _events: Map<string, Array<Event>>;
+    private _stateChanges: Array<StateChange>;
+
+    constructor(streamerMessage: StreamerMessage, executedReceipts: Receipt[], postponedReceipts: Receipt[],
+        transactions: Array<Transaction>, actions: Map<string, Action>,
+        events: Map<string, Array<Event>>, stateChanges: Array<StateChange>) {
+        this.streamerMessage = streamerMessage;
+        this.executedReceipts = executedReceipts;
+        this.postponedReceipts = postponedReceipts;
+        this.transactions = transactions;
+        this._actions = actions;
+        this._events = events;
+        this._stateChanges = stateChanges;
+    }
+
+    get blockHash(): string {
+        return this.header().hash;
+    }
+
+    get prevBlockHash(): string {
+        return this.header().prevHash;
+    }
+    get blockHeight(): number {
+        return this.header().height;
+    }
+
+    header(): BlockHeader {
+        return streamerMessageToBlockHeader(this.streamerMessage);
+    }
+
+    receipts(): Array<Receipt> {
+        if (this.executedReceipts.length == 0) {
+            this.executedReceipts = this.streamerMessage.shards.flatMap((shard) => shard.receiptExecutionOutcomes).map((executionReceipt) => outcomeWithReceiptToReceipt(executionReceipt))
+        }
+        return this.executedReceipts;
+    }
+
+    actions(): Action[] {
```

Comment:
Could someone elaborate on the difference between `Receipt[]` and `Array<Receipt>`, please? Can we use one of them everywhere?